compiler/rustc_mir/src/interpret/eval_context.rs

```diff
@@ -481,8 +481,8 @@ impl<'mir, 'tcx: 'mir, M: Machine<'mir, 'tcx>> InterpCx<'mir, 'tcx, M> {
         }
         match instance {
             ty::InstanceDef::Item(def) => {
-                if self.tcx.is_mir_available(def.did) {
-                    Ok(self.tcx.optimized_mir_opt_const_arg(def))
+                if self.tcx.is_ctfe_mir_available(def.did) {
+                    Ok(self.tcx.mir_for_ctfe_opt_const_arg(def))
```

Comment:
This means Miri-the-tool will use CTFE MIR for `const fn` but regular MIR for `fn`, right? That seems... rather confusing.

Reply:
ah yes, that is not good. I'll move mir loading to a machine method